### PR TITLE
Automated note detection and intaking using computer vision

### PR DESCRIPTION
**Pull request by having11**
Leverages the Limelight 2+ and the Coral TPU accelerator to detect notes, pick the best one, and then automatically path to/intake it

https://github.com/SubZero-Robotics/2024-CommandRobot/assets/16725165/9bfeac83-b4a5-464a-9887-2f9df259db60

### CODE REVIEW

File: src/main/cpp/utils/TargetTracker.cpp
Comment (having11) [reply src/main/cpp/utils/TargetTracker.cpp:121]: Should be fixed now

**Review by nobody5050 — COMMENTED** [review]
Will approve when tested on robot and passing ci

File: src/main/include/autos/PathFactory.h
Comment (nobody5050) [inline src/main/include/autos/PathFactory.h:78]: Should be constants

File: src/main/include/utils/Commands/DriveCommands.h
Comment (nobody5050) [inline src/main/include/utils/Commands/DriveCommands.h:96]: Commented out or removed? Should we trim this before merge?

**Review by nobody5050 — CHANGES_REQUESTED** [review]
Very good pr

File: src/main/include/utils/Commands/DriveCommands.h
Comment (having11) [reply src/main/include/utils/Commands/DriveCommands.h:96]: yeah, I can do that

**Pull request merged**
